contracts/light_client/src/lib.rs

```diff
@@ -260,12 +260,17 @@ impl LightClient {
             if let Some(approval) = block.approvals_after_next[i].clone() {
                 self.signature_set |= 1 << i;
                 self.signatures.replace(i as u64, &approval);
-                if signature_stake < this_epoch.stake_threshold {
-                    require!(self.check_block_producer_signature_in_head(i), "Signature is not valid");
+                if signature_stake < this_epoch.stake_threshold
```

Comment:
Nit: I would either remove this `signature_stake < this_epoch.stake_threshold` from if and just sum all stakes from validators with valid signatures, or break this for loop if the stake is bigger or equal to the threshold (big enough to be accepted).